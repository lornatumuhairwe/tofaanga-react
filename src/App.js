import React from 'react';
import LoginForm from './components/Login';
import './App.css';

const App = () => {
  return (
    <div>
      <h1>Welcome to Tofaanga</h1>
      <ul>
        <li><a href="/auth/login">Login</a></li>
        <li><a href="/auth/signup">Sign Up</a></li>
        <LoginForm />
      </ul>
    </div>
  );
};

export default App;
