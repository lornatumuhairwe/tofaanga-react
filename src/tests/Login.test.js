import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from '../components/Authentication/Login/Login';
import { mount } from 'enzyme';
import { stub } from 'sinon';
import App from '../App';

global.localStorage = {
  setItem: () => {}, getItem: () => {},
};

describe('LoginForm snapshot', () => {
  it('Login Form renders snapshot', () => {
    const Loginform = renderer.create(<LoginForm />);
    expect(Loginform.toJSON()).toMatchSnapshot();
  });
});

describe('Button changes the text after click', () => {
  let Login;
  const displaySignup = () => {

  };
  beforeEach(() => {
    Login = mount(<LoginForm displaySignup={displaySignup} />);
  });

  it('loads in when both parameter are put', () => {
    const loginButton = Login.find('Button');
    Login.instance().sendLogin = jest.fn();
    const loginForm = Login.find('form');
    const emailInput = loginForm.find('[type="email"]');
    const passwordInput = loginForm.find('[type="password"]');
    expect(loginButton.text()).toEqual('Sign in');
    emailInput.simulate('change', { target: { value: 'ltumuhairwe@gmail.com' } });
    passwordInput.simulate('change', { target: { value: 'test' } });
    expect(Login.state().email).toEqual('ltumuhairwe@gmail.com');
    expect(Login.state().pwd).toEqual('test');
    loginButton.simulate('submit'); // submit
    expect(Login.state().isLoading).toEqual(true);
    expect(loginButton.text()).toEqual('Loading...');
  });

  it('displays reset password form when link is clicked', () => {
    const loginForm = Login.find('form');
    const resetpasswordlink = loginForm.find('a');
    resetpasswordlink.simulate('click');
    expect(Login.state().resetpwd).toEqual(true);
  });

//   it('displays signup form when link is clicked', () => {
//     const Appi = mount(<App />);
//     console.log(Appi.state());
//     Login.instance().displaySignup = jest.fn();
//     const signuplink = Login.find('#sp');
//     signuplink.simulate('click');
//     console.log(Appi.state());
//   });
});

