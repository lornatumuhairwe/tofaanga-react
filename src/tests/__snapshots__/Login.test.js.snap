// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot Login Form renders snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-sm-6 col-md-4 col-md-offset-4"
    >
      <div
        className="notifications-wrapper"
        style={Object {}}
      />
      <h1
        className="text-center login-title"
      >
        Sign in to continue to view your bucketlists 
      </h1>
      <div
        className="account-wall"
      >
        <img
          alt=""
          className="profile-img"
          src="https://lh5.googleusercontent.com/-b0-k99FZlyE/AAAAAAAAAAI/AAAAAAAAAAA/eu7opA4byxI/photo.jpg?sz=120"
        />
        <form
          className="form-signin"
          onSubmit={[Function]}
        >
          <input
            autoFocus={true}
            className="form-control"
            id="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
            value=""
          />
          <input
            className="form-control"
            id="pwd"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
          <button
            className="btn btn-lg btn-primary btn-block btn btn-default"
            disabled={false}
            type="submit"
          >
            Sign in
          </button>
          <a
            className="pull-right need-help"
            href=""
            onClick={[Function]}
          >
            Forgot Password? 
          </a>
          <span
            className="clearfix"
          />
        </form>
      </div>
      <a
        className="text-center new-account"
        href=""
        id="sp"
        onClick={[Function]}
      >
        Create an account 
      </a>
    </div>
  </div>
</div>
`;
